actioncable/lib/action_cable/server/connection/stream.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+module ActionCable
+  module Server
+    class Connection
+      #--
+      # This class is heavily based on faye-websocket-ruby
+      #
+      # Copyright (c) 2010-2015 James Coglan
+      class Stream # :nodoc:
```

Comment:
This class also seems extremely specific to the underlying implementation - again, is this something we can replace/bypass?

Reply:
Ditto